modules/marshaller-common/src/main/java/org/apache/ignite/internal/marshaller/BinaryMode.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.marshaller;
+
+/**
+ * Various read/write modes for binary objects that maps Java types to binary types.
+ */
+public enum BinaryMode {
```

Comment:
I'm going to drop this class eventually, and replace with NativeTypeSpec.
A half done in IGNITE-15792, and the only reason it is not dropped is fixing generated marshaller out of priotiry for now.

Reply:
Ok, we can get rid of this in https://issues.apache.org/jira/browse/IGNITE-16088. I don't think anything can be done in current PR.